packages/eslint-plugin/src/rules/no-useless-template-literals.ts

```diff
@@ -83,6 +108,40 @@ export default createRule<[], MessageId>({
           context.report({
             node: expression,
             messageId: 'noUselessTemplateLiteral',
+            suggest: [
+              {
+                messageId: 'removeUselessTemplateLiteral',
+                fix(fixer): TSESLint.RuleFix[] {
+                  const index = node.expressions.indexOf(expression);
+                  const prevQuasi = node.quasis[index];
+                  const nextQuasi = node.quasis[index + 1];
+
+                  // Remove the quasis' parts that are related to the current expression.
+                  const fixes = [
+                    fixer.removeRange([
+                      prevQuasi.range[1] - 2,
+                      expression.range[0],
+                    ]),
+
+                    fixer.removeRange([
+                      expression.range[1],
+                      nextQuasi.range[0] + 1,
+                    ]),
+                  ];
+
+                  // Remove quotes for string literals (i.e. `'a'` will become `a`).
+                  const isStringLiteral =
+                    isUnderlyingTypeString(expression) &&
+                    expression.type === AST_NODE_TYPES.Literal;
+
+                  if (isStringLiteral) {
+                    fixes.push(fixer.replaceText(expression, expression.value));
+                  }
```

Comment:
We need to escape backticks here, because


```ts
`some ${'`'} string`
```

will be fixed to 

```ts
`some ` string`
```

[Playground reproduction](https://deploy-preview-8065--typescript-eslint.netlify.app/play#ts=5.3.2&fileType=.tsx&code=AYZw9gtgpgBAJAbwOTCQXxiALgJwJYB2A5sANxA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tmUSWWS18iALbF4AQ2G1Gw6BPioMkRNGgdokcGAC%2BIHUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

Reply:
Ohh, also this one:

```ts
`some ${'${aaa}'} string`
```

will be fixed to 

```ts
`some ${aaa} string`
```

[Playground reproduction](https://deploy-preview-8065--typescript-eslint.netlify.app/play#ts=5.3.2&fileType=.tsx&code=AYZw9gtgpgBAJAbwOSIIboL5IzEAXAJwEsA7Ac2CA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tmUSWWS18iALbF4AQ2G1Gw6BPioMkRNGgdokcGAC%2BIHUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

Reply:
ooh those are good ones
thanks!